### ISSUE TITLE
Removal of variable declaration statements results in rollback of all mutations in method

### ISSUE THREAD

**Issue by rouke-broersma** [stryker-mutator/stryker-net#1827]
**Describe the bug**
If you have a variable declaration without assignment you get a `Use of unassigned local variable 'xxx` exception where the variable is used. Where the variable is used is a different location than where the variable declaration was removed, resulting in all mutations in the method being removed.

**Logs**
[log-20211206.txt](https://github.com/stryker-mutator/stryker-net/files/7661172/log-20211206.txt)

**Expected behavior**
Stryker should probably not remove variable declaration statements as they will always result in compile errors anyway.

**Comment by dupdob** [comment]
Duplicate of #1745.
This is an issue with block mutator.

**Comment by rouke-broersma** [comment]
@dupdob The variable declaration that pointed to the compile errors was not a in a removed block, it was removed by the statement mutator. Are you sure it's the same issue?

**Comment by dupdob** [comment]
I may be wrong, but:

1. `use if unassigned local variable xxx`is a proof that the variable is **declared** but a not assigned.
2. in the log, I see the problem there : 
```csharp
{if(StrykerrfAmm8AcUgOY9Ob.MutantControl.IsActive(43)){}else            {
                // stuur verzoek naar rdw
                rdwHttpRequest = await _rdwClient.SendApkVerzoek(apkKeuringRequest);
            }
``` 
which appears to be a block removal mutation, as the statement mutator replaces statements by **empty** statements:
```csharp
if(StrykerrfAmm8AcUgOY9Ob.MutantControl.IsActive(45)){;}else{                _logger.LogError(e.Message);
}
```

But to be honest, I closed only after cursory looking at the log file, as I assumed this was the block mutator (as very very often)

**Comment by rouke-broersma** [comment]
Ah right, yes!
